tests/Unit/IO/Test_VolumeData.cpp

```diff
@@ -152,7 +164,40 @@ SPECTRE_TEST_CASE("Unit.IO.H5.VolumeData", "[Unit][IO][H5]") {
       grid_positions[i] =
           static_cast<size_t>(std::distance(read_grid_names.begin(), position));
     }
-
+    auto read_bases = volume_file.get_bases(observation_id);
+    std::sort(read_bases.begin(), read_bases.end(), std::less<>{});
+    auto read_quadratures = volume_file.get_quadratures(observation_id);
+    std::sort(read_quadratures.begin(), read_quadratures.end(), std::less<>{});
+    // We need non-const bases and quadratures in order to sort them, and we
+    // need them in their string form,
+    // clang-tidy consider pre-allocating the vector capacity
+    auto target_bases = [&bases]() {
+      std::vector<std::vector<std::string>> local_target_bases;
+      for (auto element_bases : bases) {
+        std::vector<std::string> axis_bases;
+        for (auto axis_basis : element_bases) {
+          axis_bases.emplace_back(MakeString{} << axis_basis);  // NOLINT
+        }
+        local_target_bases.push_back(axis_bases);
+      }
+      return local_target_bases;
+    }();
+    std::sort(target_bases.begin(), target_bases.end(), std::less<>{});
+    auto target_quadratures = [&quadratures]() {
```

Comment:
same comments as for bases

Reply:
also try to combine the lambdas by declaring once with an argument and using it twice

Reply:
@kidder or @nilsdeppe is there a standard procedure for making the one lambda work for both the bases and quadratures, doesn't seem like templates and lambdas work well together; someone on the internet suggested  `decltype`?

Reply:
You can take the argument as `const auto& basis_or_quadrature`. The `auto` is the way to get templates in lambda expressions

Reply:
Great, thanks!